Building with make fails: shell/entry.c:1: error: "_XOPEN_SOURCE" redefined [-Werror]

Full error message:
```
shell/entry.c:1: error: "_XOPEN_SOURCE" redefined [-Werror]
    1 | #define _XOPEN_SOURCE 700
      | 
<command-line>: note: this is the location of the previous definition
cc1: all warnings being treated as errors
make: *** [Makefile:44: shell/entry.o] Error 1
```

`_XOPEN_SOURCE` is defined in config.mk as 600 and redefined in shell/entry.c as 700.
The error is easy to bypass by changing shell/entry.c (removing the #define works for me, as does inserting an #undef before it), and the resulting mrsh runs fine,
but I'm unfamiliar with `_XOPEN_SOURCE` so I don't know what its value should be, or whether it should have one unique global value or different values per source file.

<details>
<summary>.build/config.mk</summary>

```
SOVERSION=0.0.0
CC=cc
PREFIX=/home/mkhl/.local
_INSTDIR=$(DESTDIR)$(PREFIX)
BINDIR?=$(_INSTDIR)/bin
LIBDIR?=$(_INSTDIR)/lib
INCDIR?=$(_INSTDIR)/include
MANDIR?=$(_INSTDIR)/share/man
PCDIR?=$(_INSTDIR)/lib/pkgconfig
CFLAGS= \
	-g \
	-std=c99 \
	-pedantic \
	-Werror \
	-Wundef \
	-Wlogical-op \
	-Wmissing-include-dirs \
	-Wold-style-definition \
	-Wpointer-arith \
	-Winit-self \
	-Wfloat-equal \
	-Wstrict-prototypes \
	-Wredundant-decls \
	-Wimplicit-fallthrough=2 \
	-Wendif-labels \
	-Wstrict-aliasing=2 \
	-Woverflow \
	-Wformat=2 \
	-Wno-missing-braces \
	-Wno-missing-field-initializers \
	-Wno-unused-parameter \
	-DHAVE_READLINE \
	-I/usr/include/readline \
	-D_DEFAULT_SOURCE \
	-D_XOPEN_SOURCE=600 \
	-U_GNU_SOURCE
LDFLAGS= \
	-fPIC \
	-Wl,--no-undefined \
	-Wl,--as-needed \
	-Wl,-soname,libmrsh.so.0 \
	-Wl,--version-script=libmrsh.gnu.sym
LIBS= \
	-lreadline
SRCDIR=.

all: mrsh highlight libmrsh.so.$(SOVERSION) $(OUTDIR)/mrsh.pc
# Begin generated rules for libmrsh
arithm.o: arithm.c
array.o: array.c
ast_print.o: ast_print.c
ast.o: ast.c
buffer.o: buffer.c
builtin/alias.o: builtin/alias.c
builtin/bg.o: builtin/bg.c
builtin/break.o: builtin/break.c
builtin/builtin.o: builtin/builtin.c
builtin/cd.o: builtin/cd.c
builtin/colon.o: builtin/colon.c
builtin/command.o: builtin/command.c
builtin/dot.o: builtin/dot.c
builtin/eval.o: builtin/eval.c
builtin/exit.o: builtin/exit.c
builtin/export.o: builtin/export.c
builtin/false.o: builtin/false.c
builtin/fg.o: builtin/fg.c
builtin/getopts.o: builtin/getopts.c
builtin/jobs.o: builtin/jobs.c
builtin/pwd.o: builtin/pwd.c
builtin/read.o: builtin/read.c
builtin/set.o: builtin/set.c
builtin/shift.o: builtin/shift.c
builtin/times.o: builtin/times.c
builtin/true.o: builtin/true.c
builtin/type.o: builtin/type.c
builtin/ulimit.o: builtin/ulimit.c
builtin/umask.o: builtin/umask.c
builtin/unalias.o: builtin/unalias.c
builtin/unset.o: builtin/unset.c
builtin/unspecified.o: builtin/unspecified.c
builtin/wait.o: builtin/wait.c
getopt.o: getopt.c
hashtable.o: hashtable.c
parser/arithm.o: parser/arithm.c
parser/parser.o: parser/parser.c
parser/program.o: parser/program.c
parser/word.o: parser/word.c
shell/arithm.o: shell/arithm.c
shell/entry.o: shell/entry.c
shell/job.o: shell/job.c
shell/path.o: shell/path.c
shell/process.o: shell/process.c
shell/redir.o: shell/redir.c
shell/shell.o: shell/shell.c
shell/task/pipeline.o: shell/task/pipeline.c
shell/task/simple_command.o: shell/task/simple_command.c
shell/task/task.o: shell/task/task.c
shell/task/word.o: shell/task/word.c
shell/word.o: shell/word.c
libmrsh_objects=\
	arithm.o \
	array.o \
	ast_print.o \
	ast.o \
	buffer.o \
	builtin/alias.o \
	builtin/bg.o \
	builtin/break.o \
	builtin/builtin.o \
	builtin/cd.o \
	builtin/colon.o \
	builtin/command.o \
	builtin/dot.o \
	builtin/eval.o \
	builtin/exit.o \
	builtin/export.o \
	builtin/false.o \
	builtin/fg.o \
	builtin/getopts.o \
	builtin/jobs.o \
	builtin/pwd.o \
	builtin/read.o \
	builtin/set.o \
	builtin/shift.o \
	builtin/times.o \
	builtin/true.o \
	builtin/type.o \
	builtin/ulimit.o \
	builtin/umask.o \
	builtin/unalias.o \
	builtin/unset.o \
	builtin/unspecified.o \
	builtin/wait.o \
	getopt.o \
	hashtable.o \
	parser/arithm.o \
	parser/parser.o \
	parser/program.o \
	parser/word.o \
	shell/arithm.o \
	shell/entry.o \
	shell/job.o \
	shell/path.o \
	shell/process.o \
	shell/redir.o \
	shell/shell.o \
	shell/task/pipeline.o \
	shell/task/simple_command.o \
	shell/task/task.o \
	shell/task/word.o \
	shell/word.o
# End generated rules for libmrsh
# Begin generated rules for mrsh
main.o: main.c
frontend/readline.o: frontend/readline.c
mrsh_objects=\
	main.o \
	frontend/readline.o
# End generated rules for mrsh
# Begin generated rules for highlight
example/highlight.o: example/highlight.c
highlight_objects=\
	example/highlight.o
# End generated rules for highlight
```

</details>

--- a/configure
+++ b/configure
@@ -252,13 +252,8 @@ fi
 
 if [ $use_readline -eq 1 ]
 then
-	append_cflags $($pkg_config --cflags $readline)
+	append_cflags $($pkg_config --cflags-only-I $readline)
 	append_libs $($pkg_config --libs $readline)
-	if [ "$readline" = "readline" ]
-	then
-		# Undo GNU_SOURCE
-		append_cflags -U_GNU_SOURCE
-	fi
 fi
 
 printf "Creating $outdir/config.mk... "

